controllers/applicationvolumereplication_controller.go

```diff
@@ -1066,6 +1121,85 @@ func (a *AVRInstance) updateAVRStatus() error {
+/*
+Description: create a new ManagedClusterView object, or update the existing one with the same name.
+Requires:
+	1) meta: specifies MangedClusterView name and managed cluster search information
+	2) viewscope: once the managed cluster is found, use this information to find the resource.
+		Optional params: Namespace, Resource, Group, Version, Kind. Resource can be used by itself, Kind requires Version
+Returns: ManagedClusterView, error
+*/
+func (r *ApplicationVolumeReplicationReconciler) getOrCreateManagedClusterView(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope) (*fndv2.ManagedClusterView, error) {
+	mcv := &fndv2.ManagedClusterView{
+		ObjectMeta: meta,
+		Spec: fndv2.ViewSpec{
+			Scope: viewscope,
+		},
+	}
+
+	err := r.Get(context.TODO(), types.NamespacedName{Name: meta.Name, Namespace: meta.Namespace}, mcv)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Create(context.TODO(), mcv)
+		}
+
+		if err != nil {
+			return nil, errorswrapper.Wrap(err, "failed to getOrCreateManagedClusterView")
+		}
+	}
+
+	if mcv.Spec.Scope != viewscope {
+		r.Log.Info("WARNING: existing ManagedClusterView has different ViewScope than desired one")
+	}
+
+	return mcv, nil
+}
+
+/*
+Description: queries a managed cluster for a resource type, and populates a variable with the results.
+Requires:
+	1) meta: information of the new/existing resource; defines which cluster(s) to search
+	2) viewscope: query information for managed cluster resource. Example: resource, name.
+	3) interface: empty variable to populate results into
+Returns: error if encountered (nil if no error occurred). See results on interface object.
+*/
+func (r *ApplicationVolumeReplicationReconciler) getManagedClusterResource(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope, resource interface{}) error {
+	// create MCV first
+	mcv, err := r.getOrCreateManagedClusterView(meta, viewscope)
+	if err != nil {
+		return errorswrapper.Wrap(err, "getManagedClusterResource failed")
+	}
+
+	// get query results
+	recentConditions := filterByConditionStatus(getMostRecentConditions(mcv.Status.Conditions), metav1.ConditionTrue)
+
+	// want single recent Condition with correct Type; otherwise: bad path
+	switch len(recentConditions) {
+	case 0:
+		err = errors.NewNotFound(schema.GroupResource{}, "failed to find a resource")
+	case 1:
+		if recentConditions[0].Type != fndv2.ConditionViewProcessing {
+			err = errors.NewNotFound(schema.GroupResource{}, "found invalid Condition.Type for ManagedClusterView results")
+		}
+	default:
+		err = fmt.Errorf("found multiple resources with ManagedClusterView - this should not be possible")
+	}
+
+	if err != nil {
+		return errorswrapper.Wrap(err, "getManagedClusterResource results")
+	}
+
+	// good path: convert raw data to usable object
+	err = json.Unmarshal(mcv.Status.Result.Raw, resource)
+	if err != nil {
+		return errorswrapper.Wrap(err, "failed to Unmarshal data from ManagedClusterView to resource")
+	}
+
+	return nil // success
+}
+
```

Comment:
I think we will need a delete interface as well, to garbage collect MCVs that we no longer need.

For example post failover once the MCV reports VRG status as Secondary in the failed cluster (preferredCluster), we would then delete the MCV at an appropriate time.

Reply:
tracked here: https://github.com/RamenDR/ramen/issues/73